src/fun_lti_provider/validator.py

```diff
@@ -0,0 +1,154 @@
+        """
+
+        try:
+            passport = LTIPassport.objects.get(
+                oauth_consumer_key=client_key, is_enabled=True
+            )
+            return passport.shared_secret
+        except LTIPassport.DoesNotExist:
+            return "dummy_client_sec_123456"
+
+    def validate_client_key(self, client_key, request):
+        """Validates that supplied client key is a registered and valid client.
+
+        Args:
+            client_key: The client/consumer key.
+            request: The calling request
+
+        Returns:
+            bool: True if the client key is registered and valid
+        """
+        try:
+            passports_count = LTIPassport.objects.filter(
+                oauth_consumer_key=client_key, is_enabled=True
+            ).count()
+            return passports_count > 0
+        except LTIPassport.DoesNotExist:
+            return False
+
+    # pylint: disable=too-many-arguments
+    def validate_timestamp_and_nonce(
+        self,
+        client_key,
+        timestamp,
+        nonce,
+        request,
+        request_token=None,
+        access_token=None,
+    ):
+        """Validates that the nonce has not been used before
+
+        Per `Section 3.3`_ of the OAuth 1.0 spec.
+
+        "A nonce is a random string, uniquely generated by the client to allow
+        the server to verify that a request has never been made before and
+        helps prevent replay attacks when requests are made over a non-secure
+        channel.  The nonce value MUST be unique across all requests with the
+        same timestamp, client credentials, and token combinations."
+
+        .. _`Section 3.3`: https://tools.ietf.org/html/rfc5849#section-3.3
+
+        One of the first validation checks that will be made is for the validity
+        of the nonce and timestamp, which are associated with a client key and
+        possibly a token. If invalid then immediately fail the request
+        by returning False. If the nonce/timestamp pair has been used before and
+        you may just have detected a replay attack. Therefore it is an essential
+        part of OAuth security that you not allow nonce/timestamp reuse.
+        Note that this validation check is done before checking the validity of
+        the client and token.::
+
+        Args:
+            client_key: The client/consumer key.
+            timestamp: The ``oauth_timestamp`` request parameter.
+            nonce: The ``oauth_nonce`` request parameter.
+            request: The calling request
+            request_token: Request token string, if any.
+            access_token: Access token string, if any.
+
+        Returns:
+            bool: True if the timestamp and nonce has not been used before
+        """
+        # pylint: disable=fixme
+        # TODO: Implement this with a redis server (to have an expiration mechanism) \
+        #  or with a database model, with a cron cleaning older entries.
+        return True
```

Comment:
use Django's cache backend?

Reply:
Thanks for your suggestion !
I'll add an implementation in this PR using django's cache backend.

Reply:
Ok, I added an [implementation](https://github.com/openfun/ashley/blob/fun_lti_provider/src/lti_provider/validator.py#L113) with django cache backend with the corresponding [tests](https://github.com/openfun/ashley/blob/fun_lti_provider/tests/lti_provider/test_lti.py#L64-L81)